pkg/bucketeer/api/client.go

```diff
@@ -15,25 +23,52 @@ type Client interface {
 
 type client struct {
 	apiKey string
+	scheme string
 	host   string
 }
 
+// Scheme is the scheme of the target service. This must be "http" or "https".
+type Scheme string
+
+func (p Scheme) Validate() bool {
+	return p == "http" || p == "https"
+}
+
 // ClientConfig is the config for Client.
 type ClientConfig struct {
 	// APIKey is the key to use the Bucketeer APIGateway service.
 	APIKey string
 
+	// Scheme is the scheme of the target service. This must be "http" or "https".
+	Scheme Scheme
+
 	// Host is the host name of the target service, e.g. api.example.com.
 	Host string
 }
 
+// Validate validates the ClientConfig.
+func (c *ClientConfig) Validate() error {
+	if c.APIKey == "" {
+		return ErrEmptyAPIKey
+	}
+	if !c.Scheme.Validate() {
+		return ErrInvalidScheme
+	}
```

Comment:
You could simplify it by doing the following.
So, you won't need to create a new public type for Scheme.

```suggestion
	if c.Scheme !=  "http" || c.Scheme != "https"{
		return ErrInvalidScheme
	}
```